File: aspnetcore/signalr/configuration/sample/hubconnectionbuilder.js
// Creates a basic connection
const connection = new signalR.HubConnectionBuilder()
    .withUrl("url")
    .build();

// Sets the transport type.
// Transport types are lsited in `HttpConnectionDispatcherOptions`
const connection = new signalR.HubConnectionBuilder()
    .withUrl("url", HttpTransportType.WebSockets)
    .build();

// Sets the protocol such as Messagepack
const connection = new signalR.HubConnectionBuilder()
    .withUrl("url")
    .withHubProtocol(IHubProtocol)
    .build();


const connection = new signalR.HubConnectionBuilder()
    .withUrl("url", IHttpConnectionOptions)
    .build();


hubConnection.serverTimeoutInMilliseconds = 5 * 1000; // 5 seconds
hubConnection.keepAliveIntervalInMilliseconds = 10 * 1000; // 10 seconds

File: aspnetcore/signalr/configuration/sample/hubconnectionbuilder.cs
// Creates a connection and restricts transports to WebSockets and Server Sent Events.
var connection = new HubConnectionBuilder()
    .WithUrl("url", HttpTransportType.WebSockets | HttpTransportType.ServerSentEvents)
    .Build();

// Creates a connection with a JSON Web Token.
var connection = new HubConnectionBuilder()
    .WithUrl("url", options => {
        options.AccessTokenProvider = async () => {
            // Get access token and return it.
        };
    })
    .Build();

// Creates a connection, and sets headers, cookies, and client certificates.
var connection = new HubConnectionBuilder()
    .WithUrl("url", options => {
        options.Headers["Foo"] = "Bar";
        options.Cookies.Add(new Cookie(...));
        options.ClientCertificates.Add(...);
    })
    .Build();